### DISCUSSION TITLE
Is there an example for child BusinessListBase in asp.net core 3.1 MVC?

### DISCUSSION THREAD

**Discussion by msrs** [MarimerLLC/csla#1772] [Questions]
Is there an example of using child BusinessListBase in asp.net core MVC application? 

I've tried looking into Csla 5.3.0 examples provided but couldn't find any. I would like to understand how model binding works with business object in MVC core application. I've issues in binding editable child list from the view to model in post operation. Esp. when I add new record to editable child list using the partial view, I couldn't set the values and post them back to controller method. 

For example, I've a editable root object person and editable child collection social media links. Each link will have two properties SocialMediaType and the LinkUrl fields along with other necessary ones like primary key field and timestamp fields. SocialMediaType is select list and LinkUrl is a textbox in the view. So, I'm having difficulties in model binding and trying to understand how to bind root and child objects to model in post back action.

Appreciated your help.

Regards,
SreeRam.

**Comment by tjscurr** [comment]
Hi @msrs,

I also had difficulties finding ASP.Net Core MVC relevant examples involving full CRUD operations for model binding of complex CSLA.Net objects utilizing the CslaModelBinder, they appear to be missing from the CSLA Samples, [although not unnoticed](https://github.com/MarimerLLC/cslaforum/issues/586).

What I have found over the last week or so, is that Razor markup in MVC Views appears to be going out of fashion in favour of Razor Pages, which apparently is [a fairly compatible technology leap, with an in project migration path](https://docs.microsoft.com/en-us/archive/msdn-magazine/2017/september/asp-net-core-simpler-asp-net-mvc-apps-with-razor-pages#migrating), and that best examples I found for both Razor Pages and Csla could be with these samples/examples:

1. [A gist from Rocky on in place editing of a root collection (Razor page with Csla)](https://gist.github.com/rockfordlhotka/a8b3253a8ee3b318ce6d30f733529075)
    - This example will help with understanding how a collection is handled with Razor pages.
2. [A project (MCM Reporting) written by Rocky](https://github.com/rockfordlhotka/mcmreporting)
     - The relevant code to your situation can be found in the solution under the "Pages/Cases" folder, specifically 
   Create.cshtml and Create.cshtml.cs, which you will see is an editable root item, with editable child collections (CaseEdit.cs)

I hope these resources help you if you choose to adopt Razor Pages, I too have just made the somewhat painful decision to abandon my older MVC experiences and come up the learning curve for Razor Pages.

Best of luck to you!
